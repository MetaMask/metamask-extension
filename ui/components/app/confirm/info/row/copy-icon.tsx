import React, { CSSProperties, useCallback } from 'react';

import { useCopyToClipboard } from '../../../../../hooks/useCopyToClipboard';
import { IconColor } from '../../../../../helpers/constants/design-system';
import {
  ButtonIcon,
  ButtonIconSize,
  IconName,
} from '../../../../component-library';

type CopyCallback = (text: string) => void;

export const CopyIcon: React.FC<{
  copyText: string;
  color?: IconColor;
  style?: CSSProperties;
}> = ({ copyText, color, style = {} }) => {
  const [copied, handleCopy] = useCopyToClipboard();

  const handleClick = useCallback(async () => {
    (handleCopy as CopyCallback)(copyText);
  }, [copyText]);

  return (
    <ButtonIcon
      color={color ?? IconColor.iconAlternative}
      iconName={copied ? IconName.CopySuccess : IconName.Copy}
      size={ButtonIconSize.Sm}
      style={{
        cursor: 'pointer',
        position: 'absolute',
        right: 0,
        top: 2,
        ...style,
      }}
      onClick={handleClick}
      ariaLabel="copy-button"
    />
  );
};
