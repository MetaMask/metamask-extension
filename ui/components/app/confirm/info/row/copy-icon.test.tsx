import React from 'react';
import { render } from '@testing-library/react';

import { CopyIcon } from './copy-icon';

describe('CopyIcon', () => {
  it('should match snapshot', () => {
    const { container } = render(<CopyIcon copyText="dummy text" />);
    expect(container).toMatchSnapshot();
  });
});
