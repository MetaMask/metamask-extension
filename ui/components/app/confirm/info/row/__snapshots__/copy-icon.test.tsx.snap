// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyIcon should match snapshot 1`] = `
<div>
  <button
    aria-label="copy-button"
    class="mm-box mm-button-icon mm-button-icon--size-sm mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-icon-alternative mm-box--background-color-transparent mm-box--rounded-lg"
    style="cursor: pointer; position: absolute; right: 0px; top: 2px;"
  >
    <span
      class="mm-box mm-icon mm-icon--size-sm mm-box--display-inline-block mm-box--color-inherit"
      style="mask-image: url('./images/icons/copy.svg');"
    />
  </button>
</div>
`;
